src/Files.App/Utils/Archives/ZipHelpers.cs

```diff
@@ -86,7 +86,17 @@ public static async Task ExtractArchive(BaseStorageFile archive, BaseStorageFold
 			int entriesFinished = 0;
 			var minimumTime = new DateTime(1);
 
-			StatusCenterItemProgressModel fsProgress = new(progress, true, FileSystemStatusCode.InProgress, entriesAmount);
+			ulong totalSize = 0;
+			foreach (var item in zipFile.ArchiveFileData)
+				totalSize += item.Size;
+
+			StatusCenterItemProgressModel fsProgress = new(
+				progress,
+				enumerationCompleted: true,
+				FileSystemStatusCode.InProgress,
+				entriesAmount,
+				(long)totalSize);
```

Comment:
Either use `long` for `totalSize` or change the type of size to `ulong` in the ViewModel.

Reply:
Each item size dat of archive is stored as ulong. so there's no way but calc with also ulong.

Reply:
Oh we would have to make it be ulong because that never be negative value